app/scripts/controllers/landing.js

```diff
@@ -28,26 +28,28 @@ angular.module('confRegistrationWebApp')
     ];
     $scope.jumboImg = jumboImgs[_.random(0, jumboImgs.length - 1)];
 
-    $scope.eventSearch = _.throttle(function(val) {
+    $scope.eventSearch = _.debounce(function(val) {
       if(!val){ return; }
       if(val.length < 2){
         return;
       }
-      $http.get('conferences', {
-        params: {
-          conferenceName: val
-        }
-      }).then(function(response){
-        $scope.eventFilters = {
-          locationName: '',
-          date: ''
-        };
-        $scope.searchComplete = val;
-        $scope.searchResults = response.data;
+      if(angular.isDefined($scope.searchVal) && val === $scope.searchVal) {
+        $http.get('conferences', {
+          params: {
+            conferenceName: val
+          }
+        }).then(function(response){
+          $scope.eventFilters = {
+            locationName: '',
+            date: ''
+          };
+          $scope.searchComplete = val;
+          $scope.searchResults = response.data;
 
-        $scope.filterUpdate();
-      });
-    }, 500, {leading: false});
+          $scope.filterUpdate();
+        });
+      }
+    }, 1000, {leading: false});
```

Comment:
I think you are right. Based on the GitHub link you posted I see leading is initially set to false.  Then later checked if options is an object and then setting to leading = !!options.leading.  I tested what would happen if I just left out the {leading: false} from the _.debounce function and seemed to operate just the same.  So I removed it.